app/src/forms/email/reminderService.js

```diff
@@ -0,0 +1,293 @@
+          late: (toDay.isBetween(dates[i].closeDate, dates[i].graceDate))? 1 : 0
+        });
+
+      }
+    }
+
+    var first = dates[0];
+    return Object({
+      state: (toDay.isBefore(first.startDate)) ? -1 : 0,
+      index: -1,
+      dates: false,
+      old_dates : false,
+      late: -1
+    });
+
+  },
+  _listDates: (schedule) =>{
+
+    if (schedule.scheduleType == ScheduleType.PERIOD){
+      return getAvailableDates(
+        schedule.keepOpenForTerm,
+        schedule.keepOpenForInterval,
+        schedule.openSubmissionDateTime,
+        schedule.repeatSubmission.everyTerm,
+        schedule.repeatSubmission.everyIntervalType,
+        schedule.allowLateSubmissions.forNext.term,
+        schedule.allowLateSubmissions.forNext.intervalType,
+        schedule.repeatSubmission.repeatUntil,
+        schedule.scheduleType,
+        schedule.closeSubmissionDateTime
+      );
+    }
+    if (schedule.scheduleType == ScheduleType.MANUAL){
+      return [Object({
+        startDate: schedule.openSubmissionDateTime,
+        closeDate:  null,
+        graceDate:  null
+      })];
+    }
+
+    if (schedule.scheduleType == ScheduleType.CLOSINGDATE){
+      return [ Object({
+        startDate: schedule.openSubmissionDateTime,
+        closeDate: schedule.closeSubmissionDateTime,
+        graceDate: service._getGraceDate(schedule)
+      })];
+    }
+
+  },
+  _getGraceDate: (schedule)=>{
+    let substartDate = moment(schedule.openSubmissionDateTime);
+    var newDate = substartDate.clone();
+    return (schedule.allowLateSubmissions.enabled) ?   newDate.add(schedule.allowLateSubmissions.forNext.term , schedule.allowLateSubmissions.forNext.intervalType).format('YYYY-MM-DD HH:MM:SS') : null ;
+  },
+  _getForms: async ()=> {
+    var fs = [];
+    await Form.query()
+      .modify('hasReminder')
+      .modify('reminderEnabled')
+      .modify('filterActive', true)
+      .then(forms => {
+        fs = forms;
+      });
+    return fs;
+  },
+  _getReminders : async (forms)=>{
+    var reminder = [];
+    var toDay = moment();
+    for (let i = 0; i< forms.length; i++) {
+      var obj = {};
+
+      obj.avalaibleDate =  service._listDates(forms[i].schedule) ;
+
+      if (obj.avalaibleDate.length == 0) {
+        reminder.push({ error:true, message : `Form ${forms[i].name } has no avalaible date.` });
+        continue;
+      }
+
+      obj.report = service.getCurrentPeriod(obj.avalaibleDate, toDay, forms[i].schedule.allowLateSubmissions.enabled);
+
+      obj.form   = forms[i];
+
+      obj.state = service._getMailType(obj.report, forms[i].reminder, forms[i].schedule.allowLateSubmissions.enabled);
+
+      if(obj.state == undefined) {
+        reminder.push({ error:true, message : ` Form ${forms[i].name } has no valid date ` });
+        continue;
+      }
+
+      await reminder.push(
+        {
+          error : false,
+          statement : exportService._getListSubmitersByFormId(forms[i].id, obj),
```

Comment:
Shouldn't the await be placed before the exportService function instead of before pushing into the array?